util/llpcElfWriter.cpp

```diff
@@ -955,6 +966,118 @@ void ElfWriter<Elf>::MergeElfBinary(
+    Reinitialize();
+    LLPC_ASSERT(relocatableElf.size() == 2 && "Can only handle VsPs Shaders for now.");
+
+    // Get the main data for the header, the parts that change will be updated when writing to buffer.
+    m_header = relocatableElf[0]->m_header;
+
+    // Copy the contents of the string table
+    ElfSectionBuffer<typename Elf::SectionHeader>* pStringTable1 = nullptr;
+    relocatableElf[0]->GetSectionDataBySectionIndex(relocatableElf[0]->m_strtabSecIdx, &pStringTable1);
+
+    ElfSectionBuffer<typename Elf::SectionHeader> *pStringTable2 = nullptr;
+    relocatableElf[1]->GetSectionDataBySectionIndex(relocatableElf[1]->m_strtabSecIdx, &pStringTable2);
+
+    MergeSection(pStringTable1,
+                 pStringTable1->secHead.sh_size,
+                 nullptr,
+                 pStringTable2,
+                 0,
+                 nullptr,
+                 &m_sections[m_strtabSecIdx]);
+
+    // Merge text sections
+    ElfSectionBuffer<typename Elf::SectionHeader>* pTextSection1 = nullptr;
+    relocatableElf[0]->GetTextSectionData(&pTextSection1);
+    ElfSectionBuffer<typename Elf::SectionHeader> *pTextSection2 = nullptr;
+    relocatableElf[1]->GetTextSectionData(&pTextSection2);
+
+    MergeSection(pTextSection1,
+                 Pow2Align(pTextSection1->secHead.sh_size, 0x100),
+                 nullptr,
+                 pTextSection2,
+                 0,
+                 nullptr,
+                 &m_sections[m_textSecIdx]);
+
+    // Build the symbol table.  First set the symbol table section header.
+    ElfSectionBuffer<typename Elf::SectionHeader>* pSymbolTableSection = nullptr;
+    relocatableElf[0]->GetSectionDataBySectionIndex(relocatableElf[0]->m_symSecIdx, &pSymbolTableSection);
+    m_sections[m_symSecIdx].secHead = pSymbolTableSection->secHead;
+
+    // Now get the symbols that belong in the symbol table.
+    uint32_t offset = 0;
+    for( const ElfReader<Elf>* elf : relocatableElf)
+    {
+        uint32_t relElfTextSectionId = elf->GetSectionIndex(TextName);
+        std::vector<ElfSymbol> symbols;
+        elf->GetSymbolsBySectionIndex(relElfTextSectionId, symbols);
+        for( auto sym : symbols) {
+            // When relocations start being used, we will have to filter out symbols related to relocations.
+            // We should be able to filter the BB* symbols as well.
+            ElfSymbol *newSym = GetSymbol(sym.pSymName);
+            newSym->secIdx = m_textSecIdx;
+            newSym->pSecName = nullptr;
+            newSym->value = sym.value + offset;
+            newSym->size = sym.size;
+            newSym->info = sym.info;
+        }
+
+        // Update the offset for the next elf file.
+        ElfSectionBuffer<typename Elf::SectionHeader>* pTextSection = nullptr;
+        elf->GetSectionDataBySectionIndex(relElfTextSectionId, &pTextSection);
+        offset += Pow2Align(pTextSection->secHead.sh_size, 0x100);
+    }
+
+    // Apply relocations
```

Comment:
I suppose this function isn't finished yet, it is just a framework. since I didn't see any code related with relocate operations.

Reply:
Yes, that is correct.  That is why I have the comment on the next line "There are no relocations to apply yet."  The implication is that there will be in the future.